core/deployment/src/main/java/io/quarkus/deployment/ExtensionLoader.java

```diff
@@ -870,7 +871,7 @@ public boolean canHandleObject(final Object obj, final boolean staticInit) {
             final Consumer<BuildStepBuilder> finalStepConfig = stepConfig.andThen(methodStepConfig)
                     .andThen(buildStepBuilder -> buildStepBuilder.buildIf(finalAddStep));
             final BiConsumer<BuildContext, Object> finalStepInstanceSetup = stepInstanceSetup;
-            final String name = clazz.getName() + "#" + method.getName();
+            final String name = clazz.getName() + "#" + method.getName() + "_" + HashUtil.methodSignatureSha1(method);
```

Comment:
I'm a bit concerned about this because IIRC until now these names were used *only* for diagnostic (human readable) messages.  Why does it need to be a hash?

Reply:
... because this `name` translates to `buildStep.toString()` and `BuildInfo.id` which I intransparently use for sorting BuildInfos.

Reply:
I understood that, but why hash it, especially given that the value is meant to be human-readable?  The hash doesn't seem to contribute anything in terms of algorithmic behavior.

Reply:
Yes, both the hash could be replaced by some string representation of the method signature here. Why I took hash: because I copy-pasted it from another commit of mine where the string is a part of a class name. The hash guarantees a lexically correct and reasonably long class name there (which is indeed not necessary here).

Reply:
No change on this place in aacdcf7